debug/org.eclipse.cdt.debug.ui/src/org/eclipse/cdt/debug/internal/ui/actions/PinCommandEnablementTester.java

```diff
@@ -0,0 +1,65 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Advantest Europe GmbH and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ * 				Raghunandana Murthappa
+ *******************************************************************************/
+package org.eclipse.cdt.debug.internal.ui.actions;
+
+import java.util.Set;
+
+import org.eclipse.core.expressions.PropertyTester;
+import org.eclipse.debug.core.DebugPlugin;
+import org.eclipse.debug.core.ILaunch;
+import org.eclipse.ui.IViewPart;
+import org.eclipse.ui.part.WorkbenchPart;
+
+/**
+ * Tests if Pin View Command can be enabled. This is tested in 2 steps.
+ * <p>
+ * 1. If a view is already pinned(which will be passes as argument) then command
+ * must be enabled until it is unpinned. 2. A valid CDT Debug Launch is active
+ * within Debug Launch Manager.
+ * </p>
+ *
+ * @author Raghunandana Murthappa
+ */
+public class PinCommandEnablementTester extends PropertyTester {
+
+	private static final String GDB_LAUNCH_TYPE = "org.eclipse.cdt.dsf.gdb.launching.GdbLaunch"; //$NON-NLS-1$
+	private static final String PIN_VIEW_COMMAND_PROP_TEST_NAME = "canPinViewEnabled"; //$NON-NLS-1$
+
+	@Override
+	public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
+		if (PIN_VIEW_COMMAND_PROP_TEST_NAME.equals(property)) {
+			// Check if the View was pinned, then button must be there to Unpin.
+			Set<IViewPart> pinnedViews = PinViewHandler.getPinnedViews();
+			if (pinnedViews.size() > 0 && args.length == 1) {
+				for (IViewPart viewPart : pinnedViews) {
+					WorkbenchPart vPart = (WorkbenchPart) viewPart;
+					if (args[0].equals(vPart.getPartName())) {
+						return true;
+					}
+				}
+			}
```

Comment:
@raghucssit what I was thinking is this, leave command visible everywhere if any pin is in use anywhere. That removes need for argument and removes current bug where it depends on whether it is primary or secondary view that the pin is applied to.

```suggestion
			// Check if any View was pinned, then leave pin visible on all views
			if (!PinViewHandler.getPinnedViews().isEmpty()) {
				return true;
			}
```

Reply:
I am working on Non declarative approach. I have registered a `IDebugContextListener` with `Debug Service`. Now I get debug changes. Based on the change I will poll the views on the workbench contribute Action dynamically.
Hope this is not an overkill..:-)

Reply:
Please see https://github.com/eclipse-cdt/cdt/pull/1049#pullrequestreview-2568048108 AFAICT the pin command shows when Java process is selected with your version because there is some gdblaunch present pin commands are visible.

Reply:
> This will enable Pin Command on all the Views if only one View is pinned.

Yes - this is intent of my recommendation, make pin commands either visible or not visible globally rather than trying to do it per view. This is a simplification that I assume would resolve the issue. i.e. pin command is only visible if user is actually doing CDT work.